custom/icds_reports/views.py

```diff
@@ -675,6 +675,16 @@ def post(self, request, *args, **kwargs):
             ).to_export('csv', location)
         elif indicator == ISSNIP_MONTHLY_REGISTER_PDF:
             awcs = request.POST.get('selected_awcs').split(',')
+            if 'all' in awcs:
+                location = request.POST.get('location', '')
+                if location:
+                    awcs = [loc.location_id for loc in SQLLocation.objects.get(
+                        location_id=location).get_descendants().filter(
+                        location_type__code=AWC_LOCATION_TYPE_CODE)]
+                else:
+                    awcs = [loc.location_id for loc in SQLLocation.objects.filter(
+                        location_type__code=AWC_LOCATION_TYPE_CODE
+                    )]
```

Comment:
Have we confirmed this is okay from a performance perspective?

Reply:
@czue This will be changed in the other PR. I got information about restrictions to the location filter in the ticket.

Reply:
FYI you can also call `.location_ids()` on a location queryset to return `queryset.values_list('location_id', flat=True)`, which will only return the ID of each location from the DB.  As implemented here, it will fetch the whole location objects from the DB.

Reply:
@esoergel Thanks, for the information about ```.location_ids()``` function.